### PR TITLE
Remove run-time dependency on ninja

### PR DESCRIPTION
**Pull request by mlxd**
**Context:** Ninja availability for lightning should be a compile time dependency only. Currently, it is listed as a runtime dependency.

**Description of the Change:** Remove ninja as a runtime dependency, and allow optional use at compile time in unavailable.

**Benefits:** Better support on platforms without spack wheels.

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,9 @@
 
 ### Improvements
 
+* Remove runtime dependench on ninja build system.
```
Comment (AmintorDusko) [inline .github/CHANGELOG.md:9]:
```suggestion
* Remove runtime dependency on ninja build system.
```

**Review by AmintorDusko — APPROVED** [review]
I have just a suggestion to fix a typo.
Good job here!

File: setup.py [hunk 2]
```diff
@@ -108,13 +112,15 @@ def build_extension(self, ext: CMakeExtension):
...
+        subprocess.check_call(
+            ["cmake", "--build", ".", "--verbose"] + build_args, cwd=self.build_temp
+        )
+
 
 with open(os.path.join("pennylane_lightning", "_version.py")) as f:
     version = f.readlines()[-1].split()[-1].strip("\"'")
 
 requirements = [
-    "ninja",
```
Comment (chaeyeunpark) [inline setup.py:117]: Are all other changes besides here from new `black` version?

Comment (mlxd) [reply setup.py:117]: Yep, everything else came with the black update.

**Review by chaeyeunpark — APPROVED** [review]
Thanks @mlxd for this update!

**Pull request merged**